### DISCUSSION TITLE
TGN: __assoc__ in LastNeighborLoader

### DISCUSSION THREAD

**Discussion by ssharpe42** [pyg-team/pytorch_geometric#2919] [Q&A]
Is there a reason you don't flush `self.__assoc__` after an insert or call and make it into `torch.empty()` again? Since you are always overwriting it anyways and it would eventually fill up over time would that be more memory efficient or is there a downside to that?

**Comment by rusty1s** [comment]
I don't think a flush is necessary, as we only need to temporally store the global to local node indices. Since `__assoc__` is a vector that is stored in memory during the whole training procedure, and there is no memory overhead in simply updating its values. Let me know if that clarifies your doubts.

**Reply by ssharpe42** [reply]
Ok maybe I was just confused how `torch.empty` works. So after the whole training procedure when `__assoc__` is filled with values, will it have the same memory requirement as it did when it was initialized with an empty tensor?

**Reply by rusty1s** [reply]
Yes. `torch.empty` will allocate memory just as every other tensor, it‘s just not pre-filled with any specific values.
